operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsRetriever.java

```diff
@@ -319,46 +401,138 @@ public WlsDomainConfig readConfig() {
+          Thread.sleep(READ_CONFIG_RETRY_MILLIS);
+        } catch (InterruptedException ex) {
+          // ignore
+        }
       }
       timeRemaining = stopTime - System.currentTimeMillis();
     }
-    if (result == null && exception != null) {
+    if (jsonResult == null && exception != null) {
       LOGGER.throwing(exception);
       throw exception;
     }
+    LOGGER.exiting(jsonResult);
+    return jsonResult;
+  }
+
+  /**
+   * Method called by the Callable that is submitted from the updateDynamicClusterSize method for updating the
+   * WLS dynamic cluster size configuration.
+   *
+   * @param wlsClusterConfig The WlsClusterConfig object of the WLS cluster whose cluster size needs to be updated. The
+   *                         caller should make sure that the cluster is a dynamic cluster.
+   * @param machineNamePrefix Prefix of names of new machines to be created
+   * @param targetClusterSize The target dynamic cluster size
+   * @return true if the request to update the cluster size is successful, false if it was not successful
+   */
+
+  private boolean doUpdateDynamicClusterSize(final WlsClusterConfig wlsClusterConfig,
+                                             final String machineNamePrefix,
+                                             final int targetClusterSize) throws Exception {
+    LOGGER.entering();
+
+    String serviceURL = connectAndGetServiceURL();
+
+    boolean result = updateDynamicClusterSizeWithServiceURL(wlsClusterConfig, machineNamePrefix,
+            targetClusterSize, httpClient, serviceURL);
+
+    LOGGER.exiting(result);
+    return result;
+  }
+
+  /**
+   * Static method to update the WebLogic dynamic cluster size configuration.
+   *
+   * @param wlsClusterConfig The WlsClusterConfig object of the WLS cluster whose cluster size needs to be updated. The
+   *                         caller should make sure that the cluster is a dynamic cluster.
+   * @param machineNamePrefix Prefix of names of new machines to be created
+   * @param targetClusterSize The target dynamic cluster size
+   * @param httpClient HttpClient object for issuing the REST request
+   * @param serviceURL service URL of the WebLogic admin server
+   *
+   * @return true if the request to update the cluster size is successful, false if it was not successful
+   */
+
+  private static boolean updateDynamicClusterSizeWithServiceURL(final WlsClusterConfig wlsClusterConfig,
+                                                                final String machineNamePrefix,
+                                                                final int targetClusterSize,
+                                                                final HttpClient httpClient,
+                                                                final String serviceURL) {
+    LOGGER.entering();
+
+    boolean result = false;
+    // Create machine(s)
+    // Commented out as we are not configuring machines for servers
```

Comment:
Remove since we're not creating machines implicitly anymore?

Reply:
Right. Will also be removing some method arguments and utility class related to machine creation.